src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -271,11 +291,18 @@ public function numHistoricalChildren()
      */
     public function numChildren($cache = true)
```

Comment:
I'm wondering if we should add an option here to get `numChildren` for LIVE here? The way this is written right now, there's a strong assumption this will only be used for Draft stage.

`prepopulate_numchildren_cache` expect a `stage` parameter, but that doesn't actually affect which stage is pre-cached, just which key the cache is stored under ... which sound kind of useless since there's no way of accessing the cache unless you actually set that parameter to `Versioned::STAGE`.

Reply:
I've focused on preserving the existing behaviour; I got rid of the $stage parameter in my subsequent patch to this (the other PR); perhaps we should roll this in here?